crates/ra_project_model/src/sysroot.rs

```diff
@@ -90,27 +79,54 @@ impl Sysroot {
     }
 }
 
-fn try_find_src_path(cargo_toml: &Path) -> Result<PathBuf> {
-    if let Ok(path) = env::var("RUST_SRC_PATH") {
-        return Ok(path.into());
-    }
+fn create_command_text(program: &str, args: &[&str]) -> String {
+    format!("{} {}", program, args.join(" "))
+}
 
-    let rustc_output = Command::new("rustc")
+fn run_command_in_cargo_dir(cargo_toml: &Path, program: &str, args: &[&str]) -> Result<Output> {
+    let output = Command::new(program)
         .current_dir(cargo_toml.parent().unwrap())
-        .args(&["--print", "sysroot"])
+        .args(args)
         .output()
-        .context("rustc --print sysroot failed")?;
-    if !rustc_output.status.success() {
-        match rustc_output.status.code() {
-            Some(code) => {
-                bail!("failed to locate sysroot: rustc --print sysroot exited with code {}", code)
-            }
-            None => bail!("failed to locate sysroot: rustc --print sysroot terminated by signal"),
+        .context(format!("{} failed", create_command_text(program, args)))?;
+    if !output.status.success() {
+        match output.status.code() {
+            Some(code) => bail!(
+                "failed to run the command: '{}' exited with code {}",
```

Comment:
I wonder if it makes sense to send a PR to rust-lang/rust with the implementation of `Dispaly` for Command

Reply:
Might be a good idea.

I have no experience on how strict their reviewers are.
They have a `Debug` impl that outputs strings like `"rustup" "component" "add" "rust-src"` currently, will they be find with another impl doing almost the same?

Reply:
I think this would require quite a lot of design back and forth, but the current situation seems quite suboptimal.

* the `Debug` impl, as is, doesn't really make much sense to me personally. It's not precise enough, as it hides the cwd and env, for eaxmple. It also is completely unclear that the thing printed is a cmd. I would expect something more derive-style

```rust
Command {
    bin: "foo",
    args: ["bar", "baz"],
    current_dir: None,
}
```

* the `Display` impl seems pretty reasonable: format the command line, escaping arguments with spaces.